feat(3448): header update

vinnyhoward opened the pull request:
## **Description**

The header is being updated according to this [Figma design](https://www.figma.com/design/aMYisczaJyEsYl1TYdcPUL/Portfolio-View?m=auto&node-id=5019-59596&t=PAdxL1bg2Mk08dSk-1) (second to last screen). 

Because `WalletAccount` is being removed, the more icon or overflow icon will be moved to the `AccountSelectorList`. Here is the [PR for more info](https://github.com/MetaMask/metamask-mobile/pull/11823)

Regression Tests Passing (11/04/24): https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/6224a4f4-1c55-43cf-a622-ec6d91f42110?tab=workflows

## **Related issues**

Ticket: [#3448](https://github.com/MetaMask/MetaMask-planning/issues/3448)

## **Manual testing steps**

1. Wallet Home Page
   - Open the account picker and switch between accounts
   - Verify functionality remains the same as before

2. Copy Address
   - Use the copy address feature in the top right
   - Confirm it works as expected

3. Network Switcher
   - Switch between different networks
   - Ensure all networks load correctly and balances update

4. Overflow (Edit Account): This is on https://github.com/MetaMask/metamask-mobile/pull/11823
   - Tap on account picker and a bottom sheet will appear 
   - Click on the overflow icon (the vertical more icon) and edit your account just like before

## **Screenshots/Recordings**

| Light  | Dark  |
|:---:|:---:|
|![light_mode_header_update](https://github.com/user-attachments/assets/39fdde2b-40a9-4119-aa93-3145fb8f7c63)|![dark_mode_header_update](https://github.com/user-attachments/assets/a5c8346b-ad89-4c6b-9e1c-f95727ec7efa)|

### **Before**

| Light  | Dark  |
|:---:|:---:|
|![light_mode_original](https://github.com/user-attachments/assets/11e473f5-3b14-454f-bd14-66e418733abb)|![dark_mode_original](https://github.com/user-attachments/assets/a50064e9-5114-498d-958b-8e2ba8fddf5c)|

### **After**

| Light  | Dark  |
|:---:|:---:|
|![light_mode_screenshot](https://github.com/user-attachments/assets/e65578ed-227a-41bb-b142-f636f81590d6)|![dark_mode_screenshot](https://github.com/user-attachments/assets/3e65ffa5-83a1-4eef-bda2-95effb32fa28)|

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

vinnyhoward committed:
feat(3447): Network Picker Component  (#11715)

## **Description**
This is NOT being merged into `main` and is being merged into the
feature branch of
[feat-header-update](https://github.com/MetaMask/metamask-mobile/tree/feat-header-update)

Updating the `NetworkPicker` design according to this
[Figma](https://www.figma.com/design/aMYisczaJyEsYl1TYdcPUL/Portfolio-View?m=auto&node-id=5019-59597).
This is just a component update only and NOT the full header redesign.
The prop of `hideNetworkName` is introduced to the component and the
default value is `false`.

We could potentially just remove that flag altogether and remove the
actual network name. Just need confirmation from DS team

## **Related issues**

Feature:
[#3447](https://github.com/MetaMask/MetaMask-planning/issues/3447)

## **Manual testing steps**

1. Goto wallet home page
2. Goto `app/components/UI/Navbar/index.js` and scroll down to
`PickerNetwork` and add prop `hideNetworkName={true}`
3. Or goto storybook

## **Screenshots/Recordings**

### Light Mode
| Regular  | Hidden  |
|:---:|:---:|

|![light_mode_regular](https://github.com/user-attachments/assets/0a889689-2d2f-4066-ba8b-355736a0228d)|![light_mode_hidden](https://github.com/user-attachments/assets/d0f0bade-9094-4973-bef6-1f4915b04b4d)|

### Dark Mode
| Regular  | Hidden  |
|:---:|:---:|

|![dark_mode_regular](https://github.com/user-attachments/assets/2a2a1fb3-4004-4715-b3eb-822ad3faf572)|![dark_mode_hidden](https://github.com/user-attachments/assets/efafd673-50a7-4bee-84ba-6553db55528d)|

### Storybook

<img
src="https://github.com/user-attachments/assets/590d5ce2-4d21-4c09-9e24-edcb7784897c"
width="320px" alt="storybook"/>

### **Before**

NA

### **After**

NA

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile
Coding
Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [x] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

vinnyhoward committed:
feat(3449): Simplified Address Copy Component (#11749)

## **Description**
This is NOT being merged into `main` and is being merged into the
feature branch of
[feat-header-update](https://github.com/MetaMask/metamask-mobile/tree/feat-header-update)

Updating the `AddressCopy` component to be the new design according to
this
[Figma](https://www.figma.com/design/aMYisczaJyEsYl1TYdcPUL/Portfolio-View?m=auto&node-id=5019-59597).
This is just a component update only and NOT the full header redesign.
The component has been simplified and there are no other affected
screens other than `WalletAccount` component

The `WalletAccount` will possibly be removed in this PR as well after
some feedback from the @MetaMask/design-system team on what they think.
It exists no where else and isn't needed

## **Related issues**

Feature:
[#3449](https://github.com/MetaMask/MetaMask-planning/issues/3449)

## **Manual testing steps**

1. Goto wallet home page (ignore the fact the wallet action component is
missing)

## **Screenshots/Recordings**


![address_copy](https://github.com/user-attachments/assets/64f76cbb-9e09-45a1-bbbd-c0e94ddbdf05)

### **Before**

NA

### **After**

NA

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile
Coding
Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [x] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

vinnyhoward committed:
update: picker account redesign

vinnyhoward committed:
update: move overflow flow

vinnyhoward committed:
Merge branch 'main' of github.com:MetaMask/metamask-mobile into feat-header-update

vinnyhoward committed:
fix: tests relating to missing mock func

vinnyhoward committed:
Merge branch 'main' into feat-header-update

vinnyhoward committed:
Merge branch 'main' into feat-header-update

vinnyhoward committed:
fix: reverting overflow related files in favor of a separate pull request for easier review

vinnyhoward committed:
Merge branch 'feat-header-update' of github.com:MetaMask/metamask-mobile into feat-header-update

vinnyhoward committed:
Merge branch 'main' into feat-header-update

vinnyhoward committed:
fix: updated ui according to feedback

vinnyhoward committed:
Merge branch 'main' of github.com:MetaMask/metamask-mobile into feat-header-update

vinnyhoward committed:
Merge branch 'feat-header-update' of github.com:MetaMask/metamask-mobile into feat-header-update

vinnyhoward committed:
fix: e2e tests relating to refs for user account

vinnyhoward committed:
Merge branch 'main' into feat-header-update

vinnyhoward committed:
Merge branch 'main' of github.com:MetaMask/metamask-mobile into feat-header-update

vinnyhoward committed:
Merge branch 'main' of github.com:MetaMask/metamask-mobile into feat-header-update

vinnyhoward committed:
Merge branch 'main' into feat-header-update

gambinish commented on app/components/hooks/useAccountName.ts:
```diff
@@ -0,0 +1,18 @@
+
+  return useMemo(
+    () =>
+      (isDefaultAccountName(defaultName) && ensName ? ensName : defaultName) ||
+      '',
```
In which situation would `isDefaultAccountName(defaultName) && ensName ? ensName : defaultName` not return a value? I think the `|| ''` might be redundant here.

gambinish replied:
```suggestion
      isDefaultAccountName(defaultName) && ensName ? ensName : defaultName)
```
☝️ not tested

vinnyhoward replied:
Added new tests for this hook

gambinish commented on app/components/UI/Navbar/index.js:
```diff
@@ -1120,7 +1155,7 @@ export function getImportTokenNavbarOptions(
         testID={ImportTokenViewSelectorsIDs.BACK_BUTTON}
       >
-        <ButtonIcon
+        {/* <ButtonIcon
```
We may want to delete this unless there is a reason for keeping the code commented out

vinnyhoward committed:
fix: remove label in picker

vinnyhoward committed:
Merge branch 'feat-header-update' of github.com:MetaMask/metamask-mobile into feat-header-update

vinnyhoward committed:
Merge branch 'main' of github.com:MetaMask/metamask-mobile into feat-header-update

gambinish commented on app/component-library/components/Pickers/PickerNetwork/PickerNetwork.test.tsx:
```diff
@@ -27,4 +29,33 @@ describe('PickerNetwork', () => {
     expect(queryByTestId(PICKERNETWORK_ARROW_TESTID)).toBeNull();
   });
+
+  it('hides network name and shows icon when hideNetworkName is true', () => {
+    const { queryByTestId } = render(
+      <PickerNetwork
+        label={SAMPLE_PICKERNETWORK_PROPS.label}
+        imageSource={SAMPLE_PICKERNETWORK_PROPS.imageSource}
+        hideNetworkName
+      />,
+    );
+
+    expect(
+      queryByTestId(WalletViewSelectorsIDs.NAVBAR_NETWORK_TEXT),
+    ).toBeNull();
+  });
```
Should we also add a test for when `hideNetworkName` is true? Or, is this sufficiently captured in the snapshot on line 17?

Admittedly, I'm not the biggest fan of snapshot tests, and usually prefer to target testIDs directly like you are doing here.

vinnyhoward replied:
Added the new. Thanks for the idea

gambinish commented on app/component-library/components/Pickers/PickerNetwork/PickerNetwork.test.tsx:
```diff
@@ -1,9 +1,11 @@
 // Third party dependencies.
 import React from 'react';
+import { fireEvent, render } from '@testing-library/react-native';
+// External dependencies.
```
Nit: Are we missing whitespace after the import?

vinnyhoward replied:
Fixed

vinnyhoward committed:
Merge branch 'main' of github.com:MetaMask/metamask-mobile into feat-header-update

vinnyhoward committed:
Merge branch 'main' into feat-header-update

vinnyhoward committed:
fix: reverting a style for picker network

vinnyhoward committed:
Merge branch 'main' of github.com:MetaMask/metamask-mobile into feat-header-update

vinnyhoward committed:
fix: added tests for new hook, updated tests ofr picker network

vinnyhoward committed:
Merge branch 'main' of github.com:MetaMask/metamask-mobile into feat-header-update

vinnyhoward committed:
Merge branch 'main' into feat-header-update

vinnyhoward committed:
feat(3481): overflow update (#11823)

## **Description**
The overflow UI is being updated as part of a bigger [redesign for the
header](https://www.figma.com/design/aMYisczaJyEsYl1TYdcPUL/Portfolio-View?m=auto&node-id=5019-59596&t=PAdxL1bg2Mk08dSk-1)

Part of this design is the removal of the `WalletAccount` component. It
will no longer be used and can probably be removed. I will confirm and
create a follow up PR to remove it.

The `WalletAccount` component will no longer be used and can probably be
removed. I will confirm and create a follow up PR. Because
`WalletAccount` is being removed, the more icon or overflow icon will be
moved to the `AccountSelectorList`. Here is the [Figma
design](https://www.figma.com/design/DFLO6VVgikalwAFyOln3Cb/Mobile-account-picker-update?node-id=34-4156&m=dev)

**NOTE:** This is being merged into my `feat-header-update` branch and
NOT `main`

## **Related issues**

Feature:
[#3481](https://github.com/MetaMask/MetaMask-planning/issues/3481)

## **Manual testing steps**

1. Open the wallet homepage.
2. Tap the account picker to open the bottom sheet.
3. In the bottom sheet, verify the following for the selected account:
    - Edit the account name and confirm changes are saved.
    - Check that tapping "View on Etherscan" opens the correct page.
    - Use "Share address" and confirm the correct address is shared.
    - Verify the correct private key is displayed 
4. If multiple accounts exist:
    - Switch between accounts and repeat step 3 for each.
5. Check overall UI
6. Verify smooth animations and proper styling.

### Changes to `RevealPrivateCredential` component
It now accepts `selectedAddress` as params to accommodate for the new UI
changes. If the params doesn't exist it will fallback to the the
selected account that is fetched from the account controller

1. Select your account
2. Goto settings from the bottom right icon
3. Click on "Security & Privacy"
4. Click on the "Reveal Secret Recovery Phrase" button and go through
the flow and verify that is your correct private key

## **Screenshots/Recordings**

### Edit Account

| Light  | Dark  |
|:---:|:---:|

|![light](https://github.com/user-attachments/assets/1b8b22d6-0149-4f06-a66f-3ae3f1e4b5f4)|![dark](https://github.com/user-attachments/assets/3c4bb694-f377-4c08-8097-efe45efa246f)

### Etherscan
| Light  | Dark  |
|:---:|:---:|

|![light_etherscan](https://github.com/user-attachments/assets/7a78345e-5ebd-4621-ab97-de692081e9a3)|![dark_ethersan](https://github.com/user-attachments/assets/e6b645fd-b937-4321-89a0-8efbd6c3bcd3)|


### **Before**

WIP

### **After**

WIP

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile
Coding
Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [x] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

gambinish commented on app/components/UI/Navbar/index.js:
```diff
@@ -904,12 +908,27 @@ export function getOfflineModalNavbar() {
+ * Function that returns the navigation options for the wallet screen.
  *
- * @returns {Object} - Corresponding navbar options containing headerTitle, headerTitle and headerTitle
+ * @param {string} selectedAddress - The currently selected Ethereum address
+ * @param {string} accountName - The name of the currently selected account
+ * @param {string} accountAvatarType - The type of avatar for the currently selected account
+ * @param {string} networkName - The name of the current network
+ * @param {Object} networkImageSource - The image source for the network icon
+ * @param {Function} onPressTitle - Callback function when the title is pressed
+ * @param {Object} navigation - The navigation object
+ * @param {Object} themeColors - The theme colors object
+ * @param {boolean} isNotificationEnabled - Whether notifications are enabled
+ * @param {boolean | null} isProfileSyncingEnabled - Whether profile syncing is enabled
+ * @param {number} unreadNotificationCount - The number of unread notifications
+ * @param {number} readNotificationCount - The number of read notifications
+ * @returns {Object} An object containing the navbar options for the wallet screen
```
Are all of these params being passed into `getWalletNavbarOptions`? I see 7 params being passed, but lmk if I'm missing something.

vinnyhoward replied:
Good catch, I need to update that

vinnyhoward replied:
Fixed

gambinish reviewed (dismissed):
Approved, just one question about some JSdoc stuff

vinnyhoward committed:
fix: updated js docs for getWalletNavbarOptions

vinnyhoward committed:
fix: updated AccountSelectorList to have dynamic styles based on multi select value

vinnyhoward committed:
fix: remove comment

vinnyhoward committed:
Merge branch 'main' of github.com:MetaMask/metamask-mobile into feat-header-update

vinnyhoward committed:
Merge branch 'main' of github.com:MetaMask/metamask-mobile into feat-header-update

vinnyhoward committed:
fix: lint

vinnyhoward committed:
Merge branch 'main' of github.com:MetaMask/metamask-mobile into feat-header-update

frankvonhoven commented on app/component-library/components/Pickers/PickerAccount/PickerAccount.styles.ts:
```diff
@@ -24,34 +23,41 @@ const styleSheet = (params: {
+    base: {
+      ...(style as ViewStyle),
+      flexDirection: 'row',
+      padding: 0,
+      borderWidth: 0,
```
does these need to be explicitly set to `0`?

vinnyhoward replied:
Yeah it as to do with the PickerBase component in the component library. This is what it looks like without it
![Screenshot 2024-10-30 at 11 45 05 AM](https://github.com/user-attachments/assets/0f061140-8ef7-433a-8e2e-9c8da554043d)

frankvonhoven commented on app/component-library/components/Pickers/PickerAccount/PickerAccount.styles.ts:
```diff
@@ -24,34 +23,41 @@ const styleSheet = (params: {
       ...cellAccountContainerStyle,
     },
     accountNameLabel: {
+      flexDirection: 'column',
```
column is default, you can remove 👍

vinnyhoward replied:
Fixed

frankvonhoven commented on app/component-library/components/Pickers/PickerAccount/PickerAccount.styles.ts:
```diff
@@ -24,34 +23,41 @@ const styleSheet = (params: {
-      borderRadius: 10,
-      borderColor: colors.border.default,
+    pickerAccountContainer: {
+      flexDirection: 'column',
```
again, default

vinnyhoward replied:
Fixed

cortisiko reviewed (dismissed):
Trigger the Regression E2E tests, once it passes feel free to post the link on the PR. I will approve once we have ✅ for regression

vinnyhoward committed:
Merge branch 'main' of github.com:MetaMask/metamask-mobile into feat-header-update

vinnyhoward committed:
Merge branch 'main' of github.com:MetaMask/metamask-mobile into feat-header-update

vinnyhoward committed:
Merge branch 'main' of github.com:MetaMask/metamask-mobile into feat-header-update

vinnyhoward committed:
fix: update e2e tests as well as ds components

vinnyhoward committed:
Merge branch 'main' of github.com:MetaMask/metamask-mobile into feat-header-update

vinnyhoward committed:
Merge branch 'main' of github.com:MetaMask/metamask-mobile into feat-header-update

vinnyhoward committed:
fix: failing android e2e test

vinnyhoward committed:
Merge branch 'main' into feat-header-update

vinnyhoward committed:
Merge branch 'main' into feat-header-update

vinnyhoward committed:
fix: failing sync test

brianacnguyen reviewed (changes requested):
Can you link the design to the show caret being moved to the account selector list?

vinnyhoward committed:
Merge branch 'main' into feat-header-update

chrisleewilcox reviewed (approved):
LGTM

brianacnguyen commented:
non-blocking. This PR could be split into smaller PRs. just a small note

chrisleewilcox reviewed (changes requested):
Will run regression tests as well for this.

vinnyhoward committed:
Merge branch 'main' into feat-header-update

chrisleewilcox commented:
> Will run regression tests as well for this.

Regression passed: https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/e7468f64-538c-4308-8c63-17b6d59905c1?tab=workflows

chrisleewilcox reviewed (dismissed):
LGTM

EtherWizard33 commented on app/component-library/components-temp/ListItemMultiSelectButton/ListItemMultiSelectButton.tsx:
```diff
@@ -44,7 +44,6 @@ const ListItemMultiSelectButton: React.FC<ListItemMultiSelectButtonProps> = ({
     isDisabled,
     isSelected,
   });
-
```
nit: I kind of like having an empty line before the return statement, but no need to change it just for this :)

but, I since it was the only change in the component library, I believe it could have spared you an extra codeowner approval from the DS team

EtherWizard33 commented on app/components/UI/AccountSelectorList/AccountSelectorList.tsx:
```diff
@@ -194,12 +219,16 @@ const AccountSelectorList = ({
+      const cellStyle: ViewStyle = {
         opacity: isLoading ? 0.5 : 1,
       };
+      if (!isMultiSelect) {
```
nit: I see these styles were already defined in the component, and it's not new to this PR, but I would think they should go in the styles file AccountSelectorList.styles.ts, and receive params.vars like isLoading and isMultiselect, for example how it's done in AvatarBase.styles.ts 

so far I like the PR, all very clear to understand and clean, with accessibilty and test ids, types, nice work :)

EtherWizard33 commented on app/components/Views/AccountActions/AccountActions.test.tsx:
```diff
@@ -144,7 +156,7 @@ describe('AccountActions', () => {
 
     expect(Share.open).toHaveBeenCalledWith({
-      message: '0xc4966c0d659d99699bfd7eb54d8fafee40e4a756',
+      message: '0xC4966c0D659D99699BFD7EB54D8fafEE40e4a756',
```
looks like the expected address was not checksummed before, and is checksummed now, is this to be expected?

vinnyhoward replied:
Yeah this one is expected it was just changed in the initial test to be checksummed. I had used the same address from above in line 34 and 41

vinnyhoward committed:
Merge branch 'main' of github.com:MetaMask/metamask-mobile into feat-header-update

vinnyhoward committed:
fix: added back in removed line

EtherWizard33 reviewed (approved):
looks good to me, good work!

chrisleewilcox reviewed (approved):
LGTM

vinnyhoward committed:
Merge branch 'main' into feat-header-update
